plugin/pkg/shoot/validator/admission.go

```diff
@@ -574,6 +590,40 @@ func (v *ValidateShoot) Admit(a admission.Attributes, o admission.ObjectInterfac
 	return nil
 }
 
+// validateMetaUpdateOnDeletedSeed validates updates to a Shoot's metadata in the case that the referenced Seed is
+// already marked for deletion. In this case it should not be allowed to modify any annotations except adding the
+// deletion confirmation annotation and to modify the Finalizers array. These operations are required to delete
+// Shoots to cleanup a Seed which is marked for deletion
+func validateMetaUpdateOnDeletedSeed(newMeta, oldMeta *metav1.ObjectMeta) error {
+	oldMeta = oldMeta.DeepCopy()
+
+	if reflect.DeepEqual(newMeta, oldMeta) {
+		return nil
+	}
+
+	if !reflect.DeepEqual(newMeta.Annotations, oldMeta.Annotations) {
+		newConfirmation, newHasConfirmation := newMeta.Annotations[common.ConfirmationDeletion]
+
+		// copy the new confirmation value to the old annotations to see if
+		// anything else was changed other than adding the confirmation annotation
+		if newHasConfirmation {
+			if oldMeta.Annotations == nil {
+				oldMeta.Annotations = make(map[string]string)
+			}
+			oldMeta.Annotations[common.ConfirmationDeletion] = newConfirmation
+		}
+
+		if !reflect.DeepEqual(newMeta.Annotations, oldMeta.Annotations) {
+			// something else than the confirmation annotation has changed -> deny the update
+			return errors.New("")
```

Comment:
Can we provide a proper error message here?

Reply:
No, actually not, because shoot and seed names would have to be passed down just for constructing the error message.
But anyways a valid point. I changed the function to return a bool indicating, if the update should be admitted, which might indeed be better style than returning an empty error, which the calling function acts upon.
Please take another look :)

Reply:
Can't we still return an error which contains the comment and just add the shoot and seed names in the calling code. 
```
if err := validateMetaUpdateOnDeletedSeed(&shoot.ObjectMeta, &oldShoot.ObjectMeta); err != nil {
   return admission.NewForbidden(a, fmt.Errorf("cannot update shoot '%s' on seed '%s' already marked for deletion: %v",  shoot.Name, seed.Name, err)
}
```

This way whoever calls the update knows why the validation failed -> `Only the confirmation.garden.sapcloud.io/deletion annotation can be changed`